beacon-chain/db/kv/state.go

```diff
@@ -598,3 +598,22 @@ func (s *Store) CleanUpDirtyStates(ctx context.Context, slotsPerArchivedPoint ty
 
 	return err
 }
+
+func (s *Store) isStateValidatorMigrationOver() bool {
+	// if flag is enabled, then always follow the new code path.
+	if featureconfig.Get().EnableHistoricalSpaceRepresentation {
+		return true
+	} else {
+		returnFlag := false
+		if err := s.db.View(func(tx *bolt.Tx) error {
+			mb := tx.Bucket(migrationsBucket)
+			if b := mb.Get(migrationStateValidatorsKey); bytes.Equal(b, migrationCompleted) {
+				returnFlag = true
+			}
+			return nil
+		}); err != nil {
+			return returnFlag
```

Comment:
We should return an error here. Otherwise we might be silently swallowing an issue.